src/scripts/components/VideoForm.js

```diff
@@ -52,13 +57,15 @@ class VideoForm extends Component<Props, Object> {
 
   render () {
     const onInputChange = this.handleInputChange
+    console.log(`current state of videoform:`)
+    console.log(this.state.id)
     return (
       <Form>
-        Editing video with id: {this.props.selectedVideo.id}
+        Editing video {this.state.id}
         <Input
           id="video-id"
           type="hidden"
-          value={this.props.selectedVideo.id}
+          value={this.state.id}
```

Comment:
What is the problem w/ just using `props`?

Reply:
I'm not sure, how found the form ahndling both state and props.
can we manage the state of the form with the props?

Reply:
It depends on what we are trying to accomplish. You could handle the form values purely w/ props, purely w/ state, or a combination of both. 

This is _probably_ what we want to do:

- in the `constructor`, set our initial state (wrt the form values) based on `props`. 
- do not update state wrt form values in `componentWillReceiveProps`
- When form values change, update the respective values in `state`
- When submitting the form, trigger an action that persists the change and when successful updates the redux store w/ the new relevant values. 

How does that sound?

Reply:
This keeps our true values in `props`, and while we are editing the form since `props` is not changing we have our source of truth maintained until the user successfully submits the changes on the form. This allows us to easily cancel our changes if the user does not submit the form, for example.